src/tests/integration/SparkLend.t.sol

```diff
@@ -0,0 +1,335 @@
+            ilk,
+            address(hub),
+            address(dss.dai),
+            address(sparkPool)
+        );
+        pool = D3MAaveV3TypePool(d3m.pool);
+        d3m.plan = D3MDeploy.deployAaveBufferPlan(
+            address(this),
+            admin,
+            adai
+        );
+        plan = D3MAaveTypeBufferPlan(d3m.plan);
+
+        // Init
+        vm.startPrank(admin);
+
+        D3MCommonConfig memory cfg = D3MCommonConfig({
+            hub: address(hub),
+            mom: address(mom),
+            ilk: ilk,
+            existingIlk: config.readBool(".existingIlk"),
+            maxLine: buffer * RAY * 1000,     // Set gap and max line to large number to avoid hitting limits
+            gap: buffer * RAY * 1000,
+            ttl: 0,
+            tau: config.readUint(".tau")
+        });
+        D3MInit.initCommon(
+            dss,
+            d3m,
+            cfg
+        );
+        D3MInit.initAavePool(
+            dss,
+            d3m,
+            cfg,
+            D3MAavePoolConfig({
+                king: config.readAddress(".king"),
+                adai: address(pool.adai()),
+                stableDebt: address(pool.stableDebt()),
+                variableDebt: address(pool.variableDebt())
+            })
+        );
+        D3MInit.initAaveBufferPlan(
+            d3m,
+            D3MAaveBufferPlanConfig({
+                buffer: buffer,
+                adai: address(pool.adai()),
+                adaiRevision: plan.adaiRevision()
+            })
+        );
+
+        vm.stopPrank();
+        
+        // Give us some DAI
+        dss.dai.setBalance(address(this), buffer * 100);
+
+        // Deposit WETH into the pool
+        uint256 amt = 1_000_000 * WAD;
+        DSTokenAbstract weth = DSTokenAbstract(dss.getIlk("ETH", "A").gem);
+        weth.setBalance(address(this), amt);
+        weth.approve(address(sparkPool), type(uint256).max);
+        dss.dai.approve(address(sparkPool), type(uint256).max);
+        sparkPool.deposit(address(weth), amt, address(this), 0);
+        
+        // Set the liquidity of DAI to 0 to simplify things
+        dss.dai.setBalance(address(adai), 0);
+
+        assertGt(getDebtCeiling(), 0);
+    }
+
+    // Helper functions
+    function getDebtCeiling() internal view returns (uint256) {
+        (,,, uint256 line,) = dss.vat.ilks(ilk);
+        return line;
+    }
+    function getDebt() internal view returns (uint256) {
+        (, uint256 art) = dss.vat.urns(ilk, address(pool));
+        return art;
+    }
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        unchecked {
+            z = x != 0 ? ((x - 1) / y) + 1 : 0;
+        }
+    }
+    function getSupplyUsed() internal view returns (uint256) {
+        PoolLike.ReserveData memory data = sparkPool.getReserveData(address(dss.dai));
+        return _divup((ATokenLike(adai).scaledTotalSupply() + uint256(data.accruedToTreasury)) * data.liquidityIndex, RAY);
+    }
```

Comment:
needs spacing between functions or they blend together

Reply:
Person preference to sometimes group closely related functions, but it's okay I'll add the lines.

Reply:
Fixed https://github.com/makerdao/dss-direct-deposit/pull/102/commits/c84bc4be70c2f8640a6a77eebc95529bd0f5a0d5

Reply:
> Person preference to sometimes group closely related functions

In alignment with Rune's goals of having "rules" that SubDAOs and outside contributors should be following on submitted code, we created the PE core guidelines and in anything not covered in there said we should defer to Solidity style guides.  In this case, the Style Guide does explicitly say we [should have spaces](https://docs.soliditylang.org/en/v0.8.17/style-guide.html#blank-lines) and not do this.